satellite/satellitedb/migrate.go

```diff
@@ -0,0 +1,328 @@
+						PRIMARY KEY ( node_id ),
+						UNIQUE ( node_id )
+					)`,
+					`CREATE TABLE IF NOT EXISTS projects (
+						id bytea NOT NULL,
+						name text NOT NULL,
+						description text NOT NULL,
+						created_at timestamp with time zone NOT NULL,
+						PRIMARY KEY ( id )
+					)`,
+					`CREATE TABLE IF NOT EXISTS users (
+						id bytea NOT NULL,
+						first_name text NOT NULL,
+						last_name text NOT NULL,
+						email text NOT NULL,
+						password_hash bytea NOT NULL,
+						status integer NOT NULL,
+						created_at timestamp with time zone NOT NULL,
+						PRIMARY KEY ( id )
+					)`,
+					`CREATE TABLE IF NOT EXISTS api_keys (
+						id bytea NOT NULL,
+						project_id bytea NOT NULL REFERENCES projects( id ) ON DELETE CASCADE,
+						key bytea NOT NULL,
+						name text NOT NULL,
+						created_at timestamp with time zone NOT NULL,
+						PRIMARY KEY ( id ),
+						UNIQUE ( key ),
+						UNIQUE ( name, project_id )
+					)`,
+					`CREATE TABLE IF NOT EXISTS project_members (
+						member_id bytea NOT NULL REFERENCES users( id ) ON DELETE CASCADE,
+						project_id bytea NOT NULL REFERENCES projects( id ) ON DELETE CASCADE,
+						created_at timestamp with time zone NOT NULL,
+						PRIMARY KEY ( member_id, project_id )
+					)`,
+				},
+			},
+			{
+				// some databases may have already this done, although the version may not match
+				Description: "Adjust table naming",
+				Version:     1,
+				Action: migrate.Func(func(log *zap.Logger, db migrate.DB, tx *sql.Tx) error {
+					hasStorageNodeID, err := postgresHasColumn(tx, "bwagreements", "storage_node_id")
+					if err != nil {
+						return ErrMigrate.Wrap(err)
+					}
+					if !hasStorageNodeID {
+						// - storage_node bytea NOT NULL,
+						// + storage_node_id bytea NOT NULL,
+						_, err := tx.Exec(`ALTER TABLE bwagreements
+							RENAME COLUMN storage_node TO storage_node_id;`)
+						if err != nil {
+							return ErrMigrate.Wrap(err)
+						}
+					}
+
+					hasUplinkID, err := postgresHasColumn(tx, "bwagreements", "uplink_id")
+					if err != nil {
+						return ErrMigrate.Wrap(err)
+					}
+					if !hasUplinkID {
+						// + uplink_id bytea NOT NULL,
+						_, err := tx.Exec(`
+							ALTER TABLE bwagreements
+								ADD COLUMN uplink_id BYTEA NOT NULL;
+						`)
+						if err != nil {
+							return ErrMigrate.Wrap(err)
+						}
+
+						rows, err := tx.Query(`SELECT data FROM bwagreements`)
+						if err != nil {
+							return ErrMigrate.Wrap(err)
+						}
+						for rows.Next() {
+							var data []byte
+							if err := rows.Scan(&data); err != nil {
+								return ErrMigrate.Wrap(errs.Combine(err, rows.Close()))
+							}
+
+							var rba pb.RenterBandwidthAllocation
+							if err := proto.Unmarshal(data, &rba); err != nil {
+								return ErrMigrate.Wrap(errs.Combine(err, rows.Close()))
+							}
+
+							_, err := tx.Exec(`UPDATE bwagreements SET uplink_id = ?`, rba.PayerAllocation.UplinkId.Bytes())
```

Comment:
oh for real? data is actually just an uplink id? ha i didn't pick up on that. i thought data was the bwagreement or something

Reply:
Well, it was bandwidth allocation. But now it’s just uplink id, not sure why it was changed that way though.

Reply:
oh oh i misread